### PR TITLE
Fix TF Longformer

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

This PR aims to fix the TF Longformer version in order to make it graph compliant. As seen offline with @patrickvonplaten `all_global_attentions` now is added in the output when `output_attentions=True`. The global attentions are filled with zeros in case `is_global_attn` is False (see line 897 in `TFLongformerSelfAttention`.

# Fix issue
#9333

### CODE REVIEW

File: src/transformers/models/longformer/modeling_tf_longformer.py [hunk 1]
```diff
@@ -1766,24 +1764,26 @@ def _pad_to_window_size(
                 )
             )
 
-            paddings = tf.constant([[0, 0], [0, padding_len]])
+        paddings = tf.convert_to_tensor([[0, 0], [0, padding_len]])
 
-            if input_ids is not None:
-                input_ids = tf.pad(input_ids, paddings, constant_values=pad_token_id)
+        if input_ids is not None:
+            input_ids = tf.pad(input_ids, paddings, constant_values=pad_token_id)
```
Comment (patrickvonplaten) [inline src/transformers/models/longformer/modeling_tf_longformer.py:1770]: When `padding_len==0`, then this won't change the `input_ids` correct?

File: src/transformers/models/longformer/modeling_tf_longformer.py [hunk 2]
```diff
@@ -2171,16 +2171,14 @@ def call(
 
         # set global attention on question tokens
         if inputs["global_attention_mask"] is None and inputs["input_ids"] is not None:
-            if inputs["input_ids"] is None:
-                logger.warning(
-                    "It is not possible to automatically generate the `global_attention_mask`. Please make sure that it is correctly set."
```
Comment (patrickvonplaten) [inline src/transformers/models/longformer/modeling_tf_longformer.py:2176]: can we leave this warning?

File: src/transformers/models/longformer/modeling_tf_longformer.py [hunk 3]
```diff
@@ -2171,16 +2171,14 @@ def call(
...
-            elif (
-                tf.where(inputs["input_ids"] == self.config.sep_token_id).shape[0] != 3 * inputs["input_ids"].shape[0]
+            if (
+                shape_list(tf.where(inputs["input_ids"] == self.config.sep_token_id))[0]
+                != 3 * shape_list(inputs["input_ids"])[0]
             ):
                 logger.warning(
                     f"There should be exactly three separator tokens: {self.config.sep_token_id} in every sample for questions answering. You might also consider to set `global_attention_mask` manually in the forward function to avoid this. This is most likely an error."
                 )
+                inputs["global_attention_mask"] = tf.fill(shape_list(inputs["input_ids"]), value=1)
```
Comment (patrickvonplaten) [inline src/transformers/models/longformer/modeling_tf_longformer.py:2181]: this doesn't look correct to me. The "default" global_attention_mask is all `0`s so I think it should be:

```python
inputs["global_attention_mask"] = tf.fill(shape_list(inputs["input_ids"]), value=0)
```

Also we could improve the warning a bit by appending a sentence like. `Disabling global attention for this forward pass...`

File: src/transformers/models/longformer/modeling_tf_longformer.py [hunk 4]
```diff
@@ -1523,8 +1523,7 @@ def call(
         training=False,
     ):
         all_hidden_states = () if output_hidden_states else None
-        all_attentions = () if output_attentions else None
-        all_global_attentions = () if (output_attentions and is_global_attn) else None
+        all_attentions = all_global_attentions = () if output_attentions else None
```
Comment (patrickvonplaten) [inline src/transformers/models/longformer/modeling_tf_longformer.py:1526]: nice!

**Review by patrickvonplaten — COMMENTED** [review]
I like the general direction of this PR! Here we should also run the slow tests to be sure nothing is broken. 

IMO, the only thing left to do is to correct the "default" global attention to all 0's instead of 1's (`global_attention_mask` is different from `attention_mask`)

File: src/transformers/models/longformer/modeling_tf_longformer.py (hunk 1 above)
Comment (jplu) [reply src/transformers/models/longformer/modeling_tf_longformer.py:1770]: Correct!

File: src/transformers/models/longformer/modeling_tf_longformer.py (hunk 2 above)
Comment (jplu) [reply src/transformers/models/longformer/modeling_tf_longformer.py:2176]: We can but here the problem is that we test if `inputs["input_ids"]` is None inside a if that already test if ` inputs["input_ids"]` is not None, this seems strange

**Comment by jplu** [comment]
I have already ran the slow tests as well and they all pass!

**Review by patrickvonplaten — APPROVED** [review]

File: src/transformers/models/longformer/modeling_tf_longformer.py (hunk 2 above)
Comment (sgugger) [reply src/transformers/models/longformer/modeling_tf_longformer.py:2176]: Agreed with @jplu here ;-)

**Review by sgugger — APPROVED** [review]
LGTM, thanks for fixing!

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @jplu!

**Pull request merged**